cli/azd/pkg/project/container_helper.go

```diff
@@ -198,7 +201,27 @@ func (ch *ContainerHelper) Credentials(
 		return nil, err
 	}
 
-	return ch.containerRegistryService.Credentials(ctx, targetResource.SubscriptionId(), loginServer)
+	var credential *azcli.DockerCredentials
+	credentialsError := retry.Do(
+		ctx,
+		retry.WithMaxDuration(90*time.Second, retry.NewExponential(1*time.Second)),
+		func(ctx context.Context) error {
+			cred, err := ch.containerRegistryService.Credentials(ctx, targetResource.SubscriptionId(), loginServer)
+			if err != nil {
+				var httpErr *azcore.ResponseError
+				if errors.As(err, &httpErr) {
+					if httpErr.StatusCode == 404 {
```

Comment:
Wondering if we should only apply to `crs.getAcrToken`, to target the specific HTTP call we see that 404s due to eventual consistency? Currently, this encompasses 404 calls to different endpoints, including listing container registries, listing credentials, AAD token exchange.

Reply:
I don't think a lower level call, and more general API, like `getAcrToken()` should retry on 404.  This API does not know if the url for the registry is wrong, or if it might still not available.

The `Credentials` method is currently where I see the clear intention of connecting to a registry, either with AcrToken or as an Admin user.   

`getAcrToken()` is called by `getTokenCredentials()` and when it returns an error, it ignores it and tries the `getAdminUserCredentials`.

What we are seeing in the issue from the customer, is that both functions are returning 404.

If we only apply the 404-retry to `getAcrToken` and we ever change the implementation to first call `getAdminUserCredentials` , we would have the issue back again, and we would need to apply the 404 retry to `findContainerRegistryByName()` or `getAdminUserCredentials()`.
And, if there is a new implementation for `Credential()` which forgets to do the 404-retry, the issue would happen again. 

I don't love the idea of trying `AcrToken` and `adminUserCredential` and having both returning 404.  I have updated the Credential implementation to fast return on 404.  I shouldn't be good if after a `404` from AcrToken, enought time happens for the adminUserCredential to find the registry and complete the login with adminUser.